Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1301 @@
+                }
+            }
+            else
+            {
+                _logger.LogDebug("Invalid or unknown network {0}.", token);
+            }
+        }
+
+        /// <summary>
+        /// Handler for network change events.
+        /// </summary>
+        /// <param name="sender">Sender.</param>
+        /// <param name="e">A <see cref="NetworkAvailabilityEventArgs"/> containing network availability information.</param>
+        private void OnNetworkAvailabilityChanged(object? sender, NetworkAvailabilityEventArgs e)
+        {
+            _logger.LogDebug("Network availability changed.");
+            OnNetworkChanged();
+        }
+
+        /// <summary>
+        /// Handler for network change events.
+        /// </summary>
+        /// <param name="sender">Sender.</param>
+        /// <param name="e">An <see cref="EventArgs"/>.</param>
+        private void OnNetworkAddressChanged(object? sender, EventArgs e)
+        {
+            _logger.LogDebug("Network address change detected.");
+            OnNetworkChanged();
+        }
+
+        /// <summary>
+        /// Async task that waits for 2 seconds before re-initialising the settings, as typically these events fire multiple times in succession.
+        /// </summary>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        private async Task OnNetworkChangeAsync()
+        {
+            try
+            {
+                await Task.Delay(2000).ConfigureAwait(false);
+                InitialiseInterfaces();
+                // Recalculate LAN caches.
+                InitialiseLAN(_configurationManager.GetNetworkConfiguration());
+
+                NetworkChanged?.Invoke(this, EventArgs.Empty);
+            }
+            finally
+            {
+                _eventfire = false;
+            }
+        }
+
+        /// <summary>
+        /// Triggers our event, and re-loads interface information.
+        /// </summary>
+        private void OnNetworkChanged()
+        {
+            lock (_eventFireLock)
+            {
+                if (!_eventfire)
+                {
+                    _logger.LogDebug("Network Address Change Event.");
+                    // As network events tend to fire one after the other only fire once every second.
+                    _eventfire = true;
+                    OnNetworkChangeAsync().GetAwaiter().GetResult();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Parses the user defined overrides into the dictionary object.
+        /// Overrides are the equivalent of localised publishedServerUrl, enabling
+        /// different addresses to be advertised over different subnets.
+        /// format is subnet=ipaddress|host|uri
+        /// when subnet = 0.0.0.0, any external address matches.
+        /// </summary>
+        private void InitialiseOverrides(NetworkConfiguration config)
+        {
+            lock (_intLock)
```

Comment:
Oversight - locking was on read/write and it was suggested that it was overkill, so I changed it to write only. Originally the initialisation was done in one go - and it was suggest it was separated to make the code read easier.

All the initialisation use the same lock as initialisation takes place at startup - and delayed at network change (which happens within a separate lock to ensure it's only fired once).

Reply:
>  locking was on read/write and it was suggested that it was overkill
That was probably me. Read locks are overkill imo because the interfaces etc. are external and we can't control them, so locking them to prevent writes won't make the interface available.